crates/rpc/rpc-eth-api/src/helpers/bitfinity_evm_rpc.rs

```diff
@@ -118,3 +165,133 @@ fn get_client(chain_spec: &ChainSpec) -> RpcResult<(&String, EthJsonRpcClient<Re
+
+    let tx = Transaction {
+        hash,
+        nonce: typed_tx.nonce(),
+        block_hash: None,
+        block_number: None,
+        transaction_index: None,
+        from,
+        to: typed_tx.to(),
+        value: typed_tx.value(),
+        gas_price: Some(typed_tx.effective_gas_price(None)),
+        gas: typed_tx.gas_limit() as _,
+        max_fee_per_gas: Some(typed_tx.max_fee_per_gas()),
+        max_priority_fee_per_gas: typed_tx.max_priority_fee_per_gas(),
+        max_fee_per_blob_gas: typed_tx.max_fee_per_blob_gas(),
+        input: typed_tx.input().clone(),
+        signature: Some(signature),
+        chain_id: typed_tx.chain_id(),
+        blob_versioned_hashes: typed_tx.blob_versioned_hashes(),
+        access_list: typed_tx.access_list().cloned(),
+        transaction_type: Some(typed_tx.transaction.tx_type().into()),
+        other: Default::default(),
+    };
+
+    Ok(Some(tx))
+}
+
+async fn transaction_from_evm_rpc(
+    chain_spec: &ChainSpec,
+    hash: B256,
+) -> RpcResult<Option<Transaction>> {
+    let (rpc_url, client) = get_client(chain_spec)?;
+    let tx = client.get_transaction_by_hash(hash.0.into()).await.map_err(|e| {
+        internal_rpc_err(format!(
+            "failed to forward eth_getTransactionByHash request to {}: {}",
+            rpc_url, e
+        ))
+    })?;
+
+    let Some(typed_tx) = tx else { return Ok(None) };
+
+    fn to_u128_or_err(val: ethereum_json_rpc_client::U256) -> RpcResult<u128> {
+        if val > ethereum_json_rpc_client::U256::from(u128::MAX) {
+            return Err(invalid_params_rpc_err("value expected to be less then u128::MAX"));
+        }
+        Ok(val.low_u128())
+    }
+
+    fn to_u64_or_err(val: ethereum_json_rpc_client::U256) -> RpcResult<u64> {
+        if val > ethereum_json_rpc_client::U256::from(u128::MAX) {
+            return Err(invalid_params_rpc_err("value expected to be less then u64::MAX"));
+        }
+        Ok(val.low_u64())
+    }
+
+    let convert_access_list = || {
+        let items = typed_tx
+            .access_list?
+            .0
+            .iter()
+            .map(|item| AccessListItem {
+                address: item.address.0.into(),
+                storage_keys: item.storage_keys.iter().map(|i| i.0.into()).collect(),
+            })
+            .collect();
+        Some(AccessList(items))
+    };
+
+    let signature = Signature {
+        r: U256::from_limbs(typed_tx.r.0),
+        s: U256::from_limbs(typed_tx.s.0),
+        v: U256::from(typed_tx.v.as_u64()),
+        y_parity: None,
+    };
+
+    let transaction_type = typed_tx
+        .transaction_type
+        .map(|v| {
+            let tx_type_u64 = v.as_u64();
+            if tx_type_u64 > u8::MAX as u64 {
+                return Err(invalid_params_rpc_err("transaction type should be less than u8::MAX"));
+            }
+            Ok(tx_type_u64 as u8)
+        })
+        .transpose()?;
+
+    let tx = Transaction {
```

Comment:
The bitfinity-evm-sdk 0.38 provides all these conversions. please upgrade to it and remove this custom code.